examples/meta/src/lib.rs

```diff
@@ -1,85 +1,49 @@
 #![no_std]
 
-use codec::{Decode, Encode};
 use gstd::{ext, msg, prelude::*};
+use gstd_meta::{meta, TypeInfo};
 
-static mut CURRENT_VALUE: u64 = 0;
+static mut MESSAGE_LOG: Vec<String> = vec![];
 
-#[derive(Debug, Encode, Decode)]
-struct MessageIn {
-    value: u64,
-    annotation: Vec<u8>,
+#[derive(TypeInfo)]
+pub struct MessageIn {
+    pub value: u64,
+    pub annotation: Vec<u8>,
 }
 
-#[derive(Debug, Encode, Decode)]
-struct MessageOut {
-    old_value: u64,
-    new_value: u64,
+#[derive(TypeInfo)]
+pub struct MessageOut {
+    pub old_value: u64,
+    pub new_value: u64,
 }
 
-#[no_mangle]
-pub unsafe extern "C" fn handle() {
-    let message_in: MessageIn = msg::load().expect("Failed to decode incoming message");
-    let old_value = CURRENT_VALUE;
-    CURRENT_VALUE += message_in.value;
-    ext::debug(&format!(
-        "Increased with annotation: {}",
-        String::from_utf8(message_in.annotation).expect("Invalid utf-8"),
-    ));
-
-    msg::reply(
-        MessageOut {
-            old_value,
-            new_value: CURRENT_VALUE,
-        },
-        1000000,
-        0,
-    )
+meta! {
+    title: "Example program with metadata",
+    input: MessageIn,
+    output: MessageOut,
+    init_input: MessageIn,
+    init_output: MessageOut
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn init() {
-    let message_in: MessageIn = msg::load().expect("Failed to decode incoming message");
-    CURRENT_VALUE = message_in.value;
-
-    msg::reply(
-        MessageOut {
-            old_value: 0,
-            new_value: CURRENT_VALUE,
-        },
-        1000000,
-        0,
-    )
-}
-
-fn return_slice<T>(slice: &[T]) -> *mut [i32; 2] {
-    Box::into_raw(Box::new([
-        slice.as_ptr() as isize as _,
-        slice.len() as isize as _,
-    ]))
-}
+pub unsafe extern "C" fn handle() {
```

Comment:
what is this handle function doing?

isn't it supposed to handle `MessageIn` message type ?

Reply:
PRs with providing library for interaction with meta data and parsing custom payloads are separated